python/rikai/parquet/writer.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Tools to write Rikai format datasets to disk
+"""
+
+import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+from rikai.mixin import ToDict
+from rikai.parquet.dataset import Dataset as RikaiDataset
+import pandas as pd
+
+__all__ = ["df_to_rikai"]
+
+
+def df_to_rikai(
+    df: pd.DataFrame, dest_path: str, schema: "pyspark.sql.types.StructType"
```

Comment:
wonder whether we can either infer the scheme, or use a framework agnostic schema ?

Reply:
there are two issues here. I think inferring the schema by default is doable (just sample a few rows and check).
spark row metadata has a lot going on so it'd be somewhat difficult to make it framework agnostic?

Reply:
we do have the code in ModelType.schema to define a schema in `string` tho. Can we extend that?

> I think inferring the schema by default is doable (just sample a few rows and check).

Our previous experience was very troublesome on schema inference tho. cc @bobingm.

Reply:
would love to learn what problems you ran into with schema inferencing (specifically for this use case i mean)